packages/jest-runtime/src/index.ts

```diff
@@ -525,6 +529,54 @@ export default class Runtime {
               // @ts-expect-error Jest uses @types/node@16. Will be fixed when updated to @types/node@20.11.0
               meta.dirname = path.dirname(meta.filename);
 
+              meta.resolve = (
+                specifier: string,
+                parent?: string | URL,
+              ): string | null => {
```

Comment:
```suggestion
              ): Promise<string> => {
```

and maybe throw instead of `null` if nothing is resolved?